cluster/cluster_test.go

```diff
@@ -149,6 +129,40 @@ func (suite *ClusterSuite) TestMemberAdd() {
 		members, err = c.MemberList(ctx)
 		require.NoError(t, err)
 		require.Equal(t, 3, len(members))
+
+		t.Run("test add node with one faulty node in three node cluster", func(t *testing.T) {
+			LPUrl, err := url.Parse("http://127.0.0.1:42380")
+			require.NoError(t, err)
+
+			LCUrl, err := url.Parse("http://127.0.0.1:42379")
+			require.NoError(t, err)
+
+			memberCfg := Config{
+				ServiceName:              "testservice3",
+				NodeName:                 "node4",
+				Port:                     4040,
+				InitialClusterClientUrls: []string{cfg.etcdConfig.LCUrls[0].String()},
+				etcdConfig:               embed.NewConfig(),
+			}
+			memberCfg.etcdConfig.Name = "node4"
+			memberCfg.etcdConfig.Dir = "tmp4"
+			memberCfg.etcdConfig.Logger = "zap"
+			memberCfg.etcdConfig.LPUrls = []url.URL{*LPUrl}
+			memberCfg.etcdConfig.LCUrls = []url.URL{*LCUrl}
+			memberCfg.etcdConfig.APUrls = []url.URL{*LPUrl}
+			memberCfg.etcdConfig.ACUrls = []url.URL{*LCUrl}
+			memberCfg.etcdConfig.ClusterState = embed.ClusterStateFlagExisting
```

Comment:
This is a great test! I think it is worth it to add a test with:

```golang
cfg.etcdConfig.StrictReconfigCheck = true
```

or

```yaml
strict-reconfig-check: false
```

for the cluster setup.

This is a decent amount of work, so if this won't regress then skip it I guess (as code changes are winding down).

Reply:
Thanks! In the tests, we've been using the `ping.yml` which ahs the `strict-reconfig-check: false` set. What test did you have in mind when setting it to true? That it won't allow for the added node?

Reply:
Yes. However after thinking about it, that is just testing an etcd feature, so nevermind.